src/components/vm/disks/diskAdd.jsx

```diff
@@ -441,8 +441,15 @@ export class AddDiskModalBody extends React.Component {
         stateDelta.format = getDefaultVolumeFormat(pool);
         if (['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(pool.type) > -1) {
             const volume = pool.volumes.find(vol => vol.name === value);
-            if (volume && volume.format)
+            if (volume && volume.format) {
                 stateDelta.format = volume.format;
+                if (volume.format === "iso") {
```

Comment:
I agree with the change, I just the a similar check already present. https://github.com/cockpit-project/cockpit-machines/blob/main/src/components/vm/disks/diskAdd.jsx#L506

The difference is that currently we check th filename suffix to be 'iso', but you check the volume.format attribute from the xml.

Reply:
the other difference is that I change disk type too, not only bus type.

Reply:
Since your check is actually more robust, maybe it worths removing the other condition I pointed you to and just keepiing only your special case handling.

Reply:
I think simple this.onValueChanged("device", "cdrom"); is enough. As I described in the [comment](https://github.com/cockpit-project/cockpit-machines/pull/625/commits/1c0ba9f601acca50f7e34ea35b86732e1e525465#r845996000), we should use scsi busType instead. Which is defaultly set [here](https://github.com/cockpit-project/cockpit-machines/blob/main/src/components/vm/disks/diskAdd.jsx#L514) behind it.